### ISSUE TITLE
[Unstable] Fail to debug the .NET Core project with "Docker .NET Core Attach (Preview)" configuration after debugging it more than twice  

### ISSUE THREAD

**Issue by v-ruizh** [microsoft/vscode-docker#2969]
**OS:** Win 10
**Build Version:** 1.13.0
**.NET Core SDK:** 5.0.203/6.0.100-preview.3.21202.5

**Repro Steps:**
1. Create a .NET Core MVC project by "dotnet new mvc --no-https".
2. F1 --> Type and select "Docker: Add Docker Files to Workspace" --> ASP.NET Core --> Linux --> Include optional Docker Compose files.
3. Right click the docker-compose.debug.yml --> Select "Compose Up".
4. Open launch.json and click "Add Configration…", and choose "Docker .NET Core Attach (Preview)".
5. Run the above configuration, select the running container from step4 and click "yes" to copy the debugger to the container.
6. Stop debugging and debug again.

**Expect:**
Succeed to debug this project again.

**Actual:**
Fail to debug the project with an error.
![image](https://user-images.githubusercontent.com/45614674/119471392-d2c9ce00-bd7b-11eb-9b91-ca1cdeb0c165.png)

**More Info:**
1. This is an **unstable** issue, please debug multiple times to reproduce this issue or change the image to ubuntu or alpine and debug twice if it does not reproduce.
2. This issue also reproduces for .NET Core Console project.

**Comment by karolz-ms** [comment]
This error code is CORDBG_E_TIMEOUT. There might be a race condition somewhere, or maybe the timeout we use is just a bit too short.

**Comment by bwateratmsft** [comment]
I haven't been able to reproduce this. @v-ruizh does this still occur?

**Comment by v-ruizh** [comment]
@bwateratmsft This issue still reproduces on the latest Docker extension build 20210728.5, you can get the details from below.
![testmvc](https://user-images.githubusercontent.com/45614674/127469506-b76a90df-e40b-42bf-beef-10deaa562dfc.gif)

**Comment by bwateratmsft** [comment]
Thanks! I was able to reproduce with that info. I found that clicking disconnect seems to leave everything in a good state (tried about 6-7 times in a row without issues):
![image](https://user-images.githubusercontent.com/36966225/127493196-7afc7263-8a28-4030-8e13-78f928f3e21b.png)

But using Run -> Stop Debugging doesn't, and leaves it broken after that:
![image](https://user-images.githubusercontent.com/36966225/127493340-775ce352-a778-429b-9242-63a11fd419fb.png)

I'm going to file a new issue for the C# extension, and close this one, as it seems to be an issue there. I found that if I do the same thing locally (no Docker) it stops the dotnet process entirely, which I don't think should be happening.
